forum/qa-theme/SnowFlat/js/selectLanguage.js

```diff
@@ -74,7 +72,10 @@ document.addEventListener("DOMContentLoaded", () => {
                 };
             }
 
-            codeBlock.addEventListener('click', clickHandler);
+            const codeBlocks = document.getElementsByClassName('cke_button__syntaxhighlight');

```

Comment:
Czy rozwiązaniem problemu jest podpięcie listenera pod każdy element, zamiast użycia _event delegation_ (jak to było dotychczas)? Może na Firefox element #cke_42 dodawany jest później i dlatego jest problem?

Reply:
Problem jest taki, że na Firefoksie ten element ma zupełnie inne ID, bo cke_41 - przynajmniej u mnie tak jest.

Reply:
A taki element chyba i tak jest tylko jeden, na wszelki wypadek jednak podpinam zdarzenie dla wszystkich takich. Obecnie, w danej chwili, na stronie pokazuje się tylko jeden edytor, ale nie wiem jak będzie w przyszłości. Jeżeli razi w oczy takie coś, to mogę to oczywiście zmienić.

Reply:
A czy ten element, który ma różne ID w zależności od przeglądarki ma jakiś inny atrybut (np. klasę), po którym można go złapać lub odnieść się do elementu wyżej i zachować _event delegation_? Ewentualnie łapać po jednym lub drugim ID (w zależności, które nie zwróci `null`)?

Reply:
Poprawiłem kod, choć nie jestem pewien, czy o to Ci chodziło. Możesz ocenić? :smile: